batcher/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+extern crate core;
+
+use std::future::Future;
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use futures_channel::mpsc::{unbounded, UnboundedSender};
+use futures_util::{future, pin_mut, StreamExt, TryStreamExt};
+use log::{debug, error, info};
+use sp1_sdk::ProverClient;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_tungstenite::tungstenite::Message;
+use aws_sdk_s3::client::Client as S3Client;
+use bytes::Bytes;
+use sha3::{Digest, Sha3_256};
+use tokio::sync::Mutex;
+
+use crate::types::VerificationData;
+
+pub mod types;
+pub mod s3;
+
+pub trait Listener {
+    fn listen(&self, address: &str) -> impl Future;
+}
+
+pub struct App {
+    s3_client: S3Client,
+    sp1_prover_client: ProverClient,
+    current_batch: Mutex<Vec<VerificationData>>,
+}
+
+const S3_BUCKET_NAME: &str = "storage.alignedlayer.com";
+
+// Implement the Listener trait for the App struct
+impl Listener for Arc<App> {
+    fn listen(&self, address: &str) -> impl Future {
+        async move {
+            // Create the event loop and TCP listener we'll accept connections on.
+            let try_socket = TcpListener::bind(address).await;
+            let listener = try_socket.expect("Failed to bind");
```

Comment:
could these lines be condensed like this?
```suggestion
            let listener = TcpListener::bind(address).await.expect("Failed to bind");
```